srml/staking/src/phragmen.rs

```diff
@@ -178,7 +179,12 @@ pub fn elect<T: Trait + 'static, FV, FN, FS>(
 				for e in &n.edges {
 					let c = &mut candidates[e.candidate_index];
 					if !c.elected && !c.approval_stake.is_zero() {
-						let temp = n.budget.saturating_mul(*n.load) / c.approval_stake;
+						let temp =
+							// Basic fixed-point shifting by 64.
+							// This will never saturate since n.budget cannot exceed u64,despite being stored in u128.
+							// Note that left-associativity in operators precedence is crucially important here.
+							n.budget.saturating_mul(SCALE_FACTOR_64) / c.approval_stake
+							* (*n.load / SCALE_FACTOR_64);
```

Comment:
would it not be more accurate to divide the whole thing by `SCALE_FACTOR_64` in the end, as it's likely to be a larger number?

Reply:
also, this line needs another level of indent